### ISSUE TITLE
Fix Click Mask "Load" button in UX

### ISSUE THREAD

**Issue by nylnook** [godotengine/godot#15524]
Follwing #14828 and fix in 50b975548dcd65a9506e7fb449d40404320c0983, several bugs remain so Click Mask can be used again.

**Issue description:**
On Antergos Linux 64b at least:
![capture d ecran de 2018-01-08 11-26-15](https://user-images.githubusercontent.com/456282/34721828-a75fe360-f544-11e7-9a75-276e22f1bb3d.png)
1. Load button do no allow to load a click mask (remain `<null>` after the file selection)
2. Using the import dock, it's mostly working, but dimensions are flipped (rotated 90°) 
3. Using the  "plus" button "Create a new resource in memory and edit it" in the Inspector dock, then select Bitmap, then provide a path and a name, then load in the click mask button, the click mask is not working: I can click everywhere in the texture surface and click respond, even in the black parts of the click mask...

**Minimal reproduction project:**
[A minimal example to test](https://github.com/godotengine/godot/files/1572105/click-masks.zip)

Thanks a lot for your great work !

**Comment by volzhs** [comment]
not sure but, isn't it fixed by #15483?

**Comment by akien-mga** [comment]
> not sure but, isn't it fixed by #15483?

Point 2. should be yes.

**Comment by nylnook** [comment]
Good catch @volzhs , I missed that commit, but I just tested again with a freshly compiled Godot, and it may fix point 2, but now I can't load it even with the Import dock (remain `<null>`) ... so I can't check: back to point 1...

**Comment by volzhs** [comment]
![screenshot from 2018-01-18 10-50-51](https://user-images.githubusercontent.com/8281454/35076832-a862352e-fc3d-11e7-9823-4f1098e0ab4c.png)

@nylnook I tested with your project.
click mask image should be imported as bitmap mask on import dock.
then it would be fine.

probably, it would be better its import setting is changed to Bitmap automatically.

**Comment by akien-mga** [comment]
AFAIK it works fine, but the UX is not so good yet. As such it's not critical to fix for 3.0, but it can definitely be improved for 3.1.

**Comment by mrcdk** [comment]
While doing this https://github.com/godotengine/godot/pull/15855 I couldn't get the mask to import or to apply to the `TextureButton` I had to do it manually via code:

```gdscript
func _ready():
	var gear_image = Image.new()
	gear_image.load("res://sprites/icons/gear_mask_2.png")
	var gear_mask = BitMap.new()
	gear_mask.create_from_image_alpha(gear_image)
	$TextureButton.texture_click_mask = gear_mask
```

**Comment by supagu** [comment]
@mrcdk does this code currently work for you? I just updated godot today and I'm also using create_from_image_alpha but I seem to get buttons I can't click on as if create_from_image_alpha is not working for me.

This is my C++ snippet:

`bool BUtil::set_click_mask_from_normal_alpha(Control* p_texture_button) {
    TextureButton* texture_button = Object::cast_to<TextureButton>(p_texture_button);
    Ref<Texture> texture = texture_button->get_normal_texture();
    String res_path = texture->get_path();
    print_line("Texture resource path: " + res_path);
    Ref<Image> image = texture->get_data();
    Ref<BitMap> mask = memnew(BitMap);
    mask->create_from_image_alpha(image);
    texture_button->set_click_mask(mask);
}`

**Comment by mrcdk** [comment]
@supagu yes, it does in latest master commit https://github.com/godotengine/godot/commit/4caadae0f15d658d464d125829039a2e35c68124 :

![https://i.imgur.com/Oan6TmI.gifv](https://i.imgur.com/Oan6TmI.gif)

**Comment by supagu** [comment]
Ah I found my problem, I was scaling my TextureButton which does not scale the mask.

**Comment by mrcdk** [comment]
That's what https://github.com/godotengine/godot/pull/15855 will solve.

**Comment by naturally-intelligent** [comment]
Also unable to set click mask from editor even after reimporting to bitmap (3.0.4).  Just remains <null> and says no error/warning, only this in console:

"Set texture_click_mask"

**Comment by reduz** [comment]
It works for me. I suggest showing your workflow on why it does not work so we can come up with a better way to improve usability. Also not a bug, changing to ehnahcement

**Comment by nylnook** [comment]
Hi,

I just tested again today with Ubuntu 18.04 64b and Godot freshly compiled (cb5a62903f9798da8b12f625cbd5a493f14e41d3), point 1 in my initial description (use Load button) is still not fixed. I'm sorry, this is not an enhancement but just a bug: this editor button is not working.
And this is blocking for the Escoria sub-project...

Thanks again !

**Comment by PetePete1984** [comment]
Chiming in, using a relatively fresh build (https://github.com/godotengine/godot/commit/c1779b1) I couldn't get click masks to work using either alpha pngs or black and white pngs. Dragging and dropping lead to the inspector field reverting to <null> in both cases, as did loading using the "Load" button.
Importing the png as a BitMap worked if I re-imported in black & white mode, but only after restarting the editor - the first attempt for each file spits out an error about invalid header / corrupt file at https://github.com/godotengine/godot/blob/0f4c30fb7166e84d93f0b2cb20d726f869155d98/scene/resources/texture.cpp#L488

**Comment by Calinou** [comment]
Can anyone reproduce this in Godot 3.2?

**Comment by Robin-Wils** [comment]
Yeah, I have reproduced the bug in Godot v3.2.1. 
I receive a error in the debugger.

![screen](https://user-images.githubusercontent.com/5617025/78455770-7f408080-76a0-11ea-88e1-972d5f61d9d8.png)

Project: [click_masks.zip](https://github.com/godotengine/godot/files/4431913/click_masks.zip)

**Comment by Robin-Wils** [comment]
Never mind. It does work if I import the image as BitMap. I misread something.  

Is there a way to automatically generate a BitMap from an existing image in code though? The transparent pixels black and the others white? 

Keeping two images of the same thing is kinda annoying, especially since I have many sprites in my project.

**Comment by KoBeWi** [comment]
Fixed in 4.0 (or probably earlier)
As mentioned many times, the image must be imported as BitMap. Then even using Load will work.

btw `create_from_image_alpha()` takes Image, so you need to import the texture as Image if you want to use it.
